music_assistant/controllers/streams.py

```
@@ -349,9 +362,12 @@ async def serve_queue_item_stream(self, request: web.Request) -> web.Response:
                     "Failed to get streamdetails for QueueItem %s: %s", queue_item_id, e
                 )
                 queue_item.available = False
-                raise web.HTTPNotFound(reason=f"No streamdetails for Queue item: {queue_item_id}")
+                raise InvalidDataError(f"No streamdetails for Queue item: {queue_item_id}")
```

sure about this? shouldnt we just raise an http error here ?

---

It was a line ball call when I did a pros and cons for the change. In the end I thought “No streamdetails” is more of a data problem (data exists but is invalid)? Thoughts?

---

yeah but this is a http request method. There you should return it as http error so the player will back off

---

Understood. I’ll fix all this up later today.

---

fixed